src/SourceBuild/content/eng/tools/PrBaselinePublisher/Publisher.cs

```diff
@@ -0,0 +1,378 @@
+            Log.LogError(errorMessage);
+            throw new InvalidOperationException(errorMessage);
+        }
+
+        var testResultsTreeResponse = await CreateTreeFromItemsAsync(testResultsTreeItems);
+        var parentTreeResponse = await CreateParentTreeAsync(testResultsTreeResponse, originalTreeResponse, originalTestResultsPath);
+
+        await CreateOrUpdatePullRequestAsync(parentTreeResponse, buildId, title, targetBranch);
+
+        return Log.GetExitCode();
+    }
+
+    // Return a dictionary using the filename without the 
+    // "Updated" prefix and anything after the first '.' as the key
+    private Dictionary<string, HashSet<string>> GetUpdatedFiles(string updatedTestsResultsPath) =>
+        Directory
+            .GetFiles(updatedTestsResultsPath, "Updated*", SearchOption.AllDirectories)
+            .GroupBy(updatedTestsFile => ParseUpdatedFileName(updatedTestsFile).Split('.')[0])
+            .ToDictionary(
+                group => group.Key,
+                group => new HashSet<string>(group)
+            );
+
+    private async Task<List<NewTreeItem>> UpdateLicenseScanFilesAsync(Dictionary<string, HashSet<string>> updatedFiles, List<NewTreeItem> tree)
+    {
+        foreach (var updatedFile in updatedFiles)
+        {
+            if (updatedFile.Key.Contains("Exclusions"))
+            {
+                // Combine the exclusions files
+                IEnumerable<string> parsedFile = Enumerable.Empty<string>();
+                foreach (var filePath in updatedFile.Value)
+                {
+                    var updatedFileLines = File.ReadAllLines(filePath);
+                    parsedFile = parsedFile.Any() ? parsedFile.Where(parsedLine => updatedFileLines.Contains(parsedLine)) : updatedFileLines;
+                }
+                string? content = parsedFile.Any() ? string.Join("\n", parsedFile) + "\n" : null;
+                string updatedFilePath = updatedFile.Key + ".txt";
+                tree = await UpdateFileAsync(tree, content, updatedFile.Key, updatedFilePath);
+            }
+            else
+            {
+                // Update the other files
+                foreach (var filePath in updatedFile.Value)
+                {
+                    var content = File.ReadAllText(filePath);
+                    if (content == DefaultLicenseBaselineContent)
+                    {
+                        content = null;
+                    }
+                    string originalFileName = Path.GetFileName(ParseUpdatedFileName(filePath));
+                    string updatedFilePath = Path.Combine("baselines/licenses", originalFileName);
+                    tree = await UpdateFileAsync(tree, content, originalFileName, updatedFilePath);
+                }
+            }
+        }
+        return tree;
+    }
+
+    private async Task<List<NewTreeItem>> UpdateSdkDiffFilesAsync(Dictionary<string, HashSet<string>> updatedFiles, List<NewTreeItem> tree)
```

Comment:
Why is the logic to deal with sdk and license basaelines so different?  Is there a way we can unify these two code paths to simplify the infra?

Reply:
I thought about trying to unify these two code paths, but, ultimately, it felt cleaner to leave them separate given all of the differences. That said, I'd be happy to abstract away some of the logic into additional functions like "UpdateLicenseExclusions", "UpdateSdkExclusions".